save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationToolsMenu.kt

```diff
@@ -57,7 +57,40 @@ external interface OrganizationToolsMenuProps : Props {
     /**
      * lambda for update projects
      */
-    var updateProjects: (MutableList<ProjectDto>) -> Unit
+    var updateProjects: (List<ProjectDto>) -> Unit
+}
+
+/**
+ * This function delete [projects] by [projects], add [oldProject] in [projects] and sorted its by comparator
+ * After that, projects and props are updated
+ *
+ * @param projects is list of the projects
+ * @param oldProject is an old project, it needs to be removed from the list
+ * @param newProject is a new project, it needs to be added to the list
+ * @param setProjects is setter to update projects
+ * @param updateProjects method from props, for changing props
+ */
+@Suppress("TYPE_ALIAS")
+private fun updateOneProjectInProjects(
+    projects: List<ProjectDto>,
+    oldProject: ProjectDto,
+    newProject: ProjectDto,
+    setProjects: StateSetter<List<ProjectDto>>,
+    updateProjects: (List<ProjectDto>) -> Unit
+) {
+    val comparator: Comparator<ProjectDto> =
+            compareBy<ProjectDto> { orderedProjectStatus[it.status] }
+                .thenBy { it.name }
+    projects
+        .minus(oldProject)
+        .plus(newProject)
+        .sortedWith(comparator)
+        .also {
+            setProjects(it)
+        }
+        .also {
+            updateProjects(it)
```

Comment:
These two also calls can be merged together:

```suggestion
        .also { sortedProjects ->
            setProjects(sortedProjects)
            updateProjects(sortedProjects)
```

Reply:
Done [13603f1](https://github.com/saveourtool/save-cloud/pull/1546/commits/13603f16b4f13604e28a152d433dc1ade897cb6b)